extensions/tracing/src/main/java/io/opentelemetry/extension/tracing/ExtendedTracer.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.extension.tracing;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Scope;
+import java.util.concurrent.Callable;
+
+/** Provides easy mechanisms for wrapping standard Java constructs with an OpenTelemetry Span. */
+@SuppressWarnings("InconsistentOverloads")
+public class ExtendedTracer {
+
+  private final Tracer tracer;
+
+  /** Create a new {@link ExtendedTracer} that wraps the provided Tracer. */
+  public ExtendedTracer(Tracer tracer) {
+    this.tracer = tracer;
+  }
+
+  /** Wrap the provided {@link Runnable} with a {@link Span} with the provided name. */
+  public Runnable wrap(String spanName, Runnable runnable) {
+    return wrap(tracer, spanName, runnable);
+  }
+
+  /**
+   * Wrap the provided {@link Runnable} with a {@link Span} with the provided name. Uses the
+   * provided {@link Tracer} for the functionality.
+   */
+  public static Runnable wrap(Tracer tracer, String spanName, Runnable runnable) {
+    return () -> {
+      Span span = tracer.spanBuilder(spanName).startSpan();
+      try (Scope scope = span.makeCurrent()) {
+        runnable.run();
+      } catch (Throwable e) {
+        span.recordException(e);
+        throw e;
+      } finally {
+        span.end();
+      }
+    };
+  }
```

Comment:
If you do the implementation of the Tracer then users who wants to use this will do:

```java
// For extended
private final ExtendedTracer tracer = ExtendedTracer.create(tracerProvider.get("foo"));

// Before
private final Tracer tracer = tracerProvider.get("foo");
```

So probably the static methods are not very useful.

Reply:
ok, will remove them for now.